### ISSUE TITLE
[java] UnnecessaryFullyQualifiedName: False positive for lombok annotation

### ISSUE THREAD

**Issue by altaiezior** [pmd/pmd#6731]
**Affects PMD Version:** 7.24.0

**Rule:**

Please provide the rule name and a link to the rule documentation:
https://pmd.github.io/pmd/pmd_rules_java_codestyle.html#unnecessaryfullyqualifiedname

**Description:**

**Code Sample demonstrating the issue:**

```java
@Value
@Builder(builderClassName='Builder')
class A {
    @lombk.Builder.Default
    String value = ""
```

**Expected outcome:**

PMD reports a violation at line ..., but that's wrong. That's a false positive.

Here since my builder class name is `Builder` I am being reported that I should not use fully qualified class name but I don't have any other option. So this becomes a false postive and can be corrected

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]* maven
